package/MDAnalysis/analysis/nuclinfo.py

```diff
@@ -336,24 +364,45 @@ def tors(universe, seg, i):
-                             " atom {0!s} {1!s} C3\' ".format(seg, i), " atom {0!s} {1!s} O3\' ".format(seg, i))
-    e = universe.select_atoms(" atom {0!s} {1!s} C4\' ".format(seg, i), " atom {0!s} {1!s} C3\' ".format(seg, i),
-                             " atom {0!s} {1!s} O3\' ".format(seg, i), " atom {0!s} {1!s} P    ".format(seg, i + 1))
-    z = universe.select_atoms(" atom {0!s} {1!s} C3\' ".format(seg, i), " atom {0!s} {1!s} O3\' ".format(seg, i),
-                             " atom {0!s} {1!s} P    ".format(seg, i + 1), " atom {0!s} {1!s} O5\' ".format(seg, i + 1))
+    a = universe.select_atoms(" atom {0!s} {1!s} O3\' ".format(seg, i - 1),
+                              " atom {0!s} {1!s} P  ".format(seg, i),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i))
+
+    b = universe.select_atoms(" atom {0!s} {1!s} P    ".format(seg, i),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i))
+
+    g = universe.select_atoms(" atom {0!s} {1!s} O5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i))
+
+    d = universe.select_atoms(" atom {0!s} {1!s} C5\' ".format(seg, i),
+                              " atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i))
+
+    e = universe.select_atoms(" atom {0!s} {1!s} C4\' ".format(seg, i),
+                              " atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i),
+                              " atom {0!s} {1!s} P    ".format(seg, i + 1))
+
+    z = universe.select_atoms(" atom {0!s} {1!s} C3\' ".format(seg, i),
+                              " atom {0!s} {1!s} O3\' ".format(seg, i),
+                              " atom {0!s} {1!s} P    ".format(seg, i + 1),
+                              " atom {0!s} {1!s} O5\' ".format(seg, i + 1))
     try:
-        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i), " atom {0!s} {1!s} C1\' ".format(seg, i),
-                                 " atom {0!s} {1!s} N1 ".format(seg, i), " atom {0!s} {1!s} C2  ".format(seg, i))
+        c = universe.select_atoms(" atom {0!s} {1!s} O4\' ".format(seg, i),
+                                  " atom {0!s} {1!s} C1\' ".format(seg, i),
+                                  " atom {0!s} {1!s} N1 ".format(seg, i),
+                                  " atom {0!s} {1!s} C2  ".format(seg, i))
     except:
```

Comment:
I *think* that maybe Atom selections are meant to raise an Error if they're len 0.. I just tried a bad one and I got an index error, which isn't ideal, but is handily caught by this except block.  It should probably raise a SelectionError rather than returning an empty AtomGroup though?  Although all other zero lengths selections just return empty groups.  Not entirely sure here